documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,301 @@
+[source,shell]
+----
+docker exec -it local-cassandra-instance cqlsh -e "CREATE TABLE IF NOT EXISTS k1.fruit(id text, name text, description text, PRIMARY KEY((id), name))"
+----
+
+For the local Cassandra that is not running within docker, you need to execute cqlsh commands:
+[source,shell]
+----
+cqlsh -e "CREATE KEYSPACE IF NOT EXISTS k1 WITH replication = {'class':'SimpleStrategy', 'replication_factor':1}
+cqlsh -e "CREATE TABLE IF NOT EXISTS k1.fruit(id text, name text, description text, PRIMARY KEY((id), name))
+----
+
+== Creating a frontend
+
+Now let's add a simple web page to interact with our `FruitResource`.
+Quarkus automatically serves static resources located under the `META-INF/resources` directory.
+In the `src/main/resources/META-INF/resources` directory, add a `fruits.html` file with the content from this {fruits-html-url}[fruits.html] file in it.
+
+You can now interact with your REST service:
+
+* start Quarkus with `mvn clean quarkus:dev`
+* open a browser to `http://localhost:8080/fruits.html`
+* add new fruits to the list via the form
+
+== Connection Health Check
+
+If you are using the `quarkus-smallrye-health` extension, `cassandra-quarkus` will automatically add a readiness health check
+to validate the connection to the cluster.
+
+So when you access the `/health/ready` endpoint of your application you will have information about the connection validation status.
+
+This behavior can be disabled by setting the `quarkus.cassandra.health.enabled` property to `false` in your `application.properties`.
+
+== Metrics
+
+If you are using the `quarkus-smallrye-metrics` extension, `cassandra-quarkus` can provide metrics about CqlSession and Cassandra nodes.
+This behavior must first be enabled by setting the `quarkus.cassandra.metrics.enabled` property to `true` in your `application.properties`.
+
+The next step that you need to do is set explicitly which metrics should be enabled.
+The `quarkus.cassandra.metrics.session-enabled` and `quarkus.cassandra.metrics.node-enabled` should be used for enabling metrics.
+So for example to enable `session.connected-nodes` and `node.pool.open-connections` you should add the following settings to your
+`application.properties`:
+[source, properties]
+----
+quarkus.cassandra.metrics.enabled=true
+quarkus.cassandra.metrics.session-enabled=connected-nodes
+quarkus.cassandra.metrics.node-enabled=pool.open-connections
+----
+
+For the full list of available metrics, please refer to the {reference-conf-url}[reference.conf] and the `advanced.metrics` section.
+
+So when you access the `/metrics` endpoint of your application you will have information about added metrics.
+
+== Building a native executable
+
+You can use the Cassandra client in a native executable.
+
+You can build a native executable with the `mvn clean package -Pnative` command.
+
+Running it is as simple as executing `./target/quickstart-1.0.0-SNAPSHOT-runner`.
```

Comment:
We've been describing this as creating a new app which happens to have the same content as the quickstart app rather than using the quickstart app itself.  Perhaps we should just tell the user to look for an app ending in "-runner" in the "target" directory here?

Reply:
When the user will follow this tutorial, he should copy the `pom.xml` from the quickstart. If he will do it, the name of the app will be quickstart - so the commands are proper